lib/verbalize/build.rb

```diff
@@ -34,18 +35,19 @@ def initialize(#{declaration_arguments_string})
       CODE
     end
 
-    attr_reader :required_keywords, :optional_keywords
+    attr_reader :required_keywords, :optional_keywords, :default_keywords
 
     private
 
     def all_keywords
-      required_keywords + optional_keywords
+      required_keywords + optional_keywords + default_keywords
     end
 
     def declaration_arguments_string
-      required_segments = required_keywords.map { |keyword| "#{keyword}:" }
-      optional_segments = optional_keywords.map { |keyword| "#{keyword}: nil" }
-      (required_segments + optional_segments).join(', ')
+      required_segments  = required_keywords.map { |keyword| "#{keyword}:" }
+      optional_segments  = optional_keywords.map { |keyword| "#{keyword}: nil" }
+      default_segments   = default_keywords.map  { |keyword| "#{keyword}: self.defaults[:#{keyword}].call" }
```

Comment:
Not really feedback for this PR, just some thinking about potential future changes. 

It's interesting that optional & default segments are basically the same thing. I think we could simplify some of this feature implementation by treating optional the same as default by wrapping nil in a lambda, e.g. `self.defaults[:some_input_without_a_default] => lamda { nil }`.

But that would probably hurt performance slightly for the sake of simplifying implementation. It would be interesting to see how this would effect performance.